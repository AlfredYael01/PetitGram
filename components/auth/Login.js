import React, { useState } from 'react';
import { Button, Text, TextInput, View, StyleSheet, Image } from 'react-native';
import { onAuthStateChanged, getAuth, signInWithEmailAndPassword } from 'firebase/auth';
import { Provider, useSelector, useDispatch } from 'react-redux';
import store from '../redux/store';
import { setUser } from '../redux/userActions';

export default function Login(props) {
    const [email, setEmail] = useState('');
    const [password, setPassword] = useState('');
    const [error, setError] = useState('');

    const dispatch = useDispatch();

    const onSignUp = () => {
        // Create user with email and password if not exists
        const auth = getAuth();

        signInWithEmailAndPassword(auth, email, password)
            .then((userCredential) => {
                // Signed in
                const user = userCredential.user;
                dispatch(setUser(user.uid)); // Dispatch user data to Redux
            })
            .catch((error) => {
                const errorCode = error.code;
                const errorMessage = error.message;
                switch (errorCode) {
                    case 'auth/invalid-email':
                        setError('Invalid email address format.');
                        break;
                    case 'auth/user-disabled':
                        setError('User with this email has been disabled.');
                        break;
                    case 'auth/user-not-found':
                        setError('User with this email does not exist.');
                        break;
                    case 'auth/wrong-password':
                        setError('Invalid password.');
                        break;
                    default:
                        setError('Something went wrong.');
                        break;
                }
            });

        onAuthStateChanged(auth, (user) => {
            if (user) {
                dispatch(setUser(user.uid)); // Dispatch user data to Redux
            }
        });
    }

    return (
        <View style={styles.container}>
            <Image source={require('../../assets/logo.png')} style={styles.logo} />
            <View style={styles.form}>
                <TextInput
                    style={styles.textInput}
                    placeholder="Email"
                    onChangeText={(email) => setEmail(email)}
                />
                <TextInput
                    style={styles.textInput}
                    placeholder="Password"
                    secureTextEntry={true}
                    onChangeText={(password) => setPassword(password)}
                />

                {error ? <Text style={styles.errorText}>{error}</Text> : null}

                <Button
                    style={styles.button}
                    onPress={() => onSignUp()}
                    title="Sign In"
                />
            </View>

            <Text
                style={styles.registerLink}
                onPress={() => props.navigation.navigate("Register")}
            >
                Don't have an account? Sign Up.
            </Text>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: '#F8F8F8', // Background color
    },
    logo: {
        marginBottom: 0,
        width: 250,
        height: 250,
    },
    form: {
        width: '80%',
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: 'white', // Form background color
        padding: 20,
        borderRadius: 20, // Rounded corners
        shadowColor: 'rgba(0, 0, 0, 0.2)',
        shadowOffset: { width: 0, height: 2 },
        shadowRadius: 2,
        shadowOpacity: 1,
        elevation: 2,
    },
    textInput: {
        width: '100%',
        height: 40,
        marginVertical: 10,
        padding: 10,
        borderWidth: 1,
        borderRadius: 5,
    },
    button: {
        width: '100%',
        height: 40,
        marginTop: 20,
        borderRadius: 5,
        backgroundColor: '#007BFF', // Button background color
    },
    errorText: {
        color: 'red',
        marginVertical: 10,
        textAlign: 'center',
    },
    registerLink: {
        marginTop: 20,
        color: '#007BFF', // Link color
        textDecorationLine: 'underline',
    },
});
