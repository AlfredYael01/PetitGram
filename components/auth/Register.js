import React, { useState } from 'react';
import { TouchableOpacity, Text, TextInput, View, StyleSheet, Image } from 'react-native';
import { getAuth, createUserWithEmailAndPassword } from 'firebase/auth';
import { getFirestore, collection, addDoc, getDocs, query, where} from 'firebase/firestore';
import { getStorage, ref, uploadBytes, getDownloadURL } from 'firebase/storage';
import * as ImagePicker from 'expo-image-picker';

export default Register = ({ navigation }) => {
    const defaultPhoto = 'https://firebasestorage.googleapis.com/v0/b/petitgram-b48fd.appspot.com/o/Profile%2FuserImage.png?alt=media&token=29660ffe-caba-4fe6-b028-09af3f446b74&_gl=1*i459ow*_ga*NDMzMjcyMjA3LjE2OTU4ODMxMjk.*_ga_CW55HF8NVT*MTY5OTM0NzExOC4xOS4xLjE2OTkzNDcxNzIuNi4wLjA';
    const [email, setEmail] = useState('');
    const [password, setPassword] = useState('');
    const [validatePassword, setValidatePassword] = useState('');
    const [pseudo, setPseudo] = useState('');
    const [name, setName] = useState('');
    const [description, setDescription] = useState('');
    const [photo, setPhoto] = useState(defaultPhoto);
    const [error, setError] = useState('');

    const checkForEmptyFields = () => {
        if (email === '' || password === '') {
            setError('Email and password fields cannot be empty.');
            return true;
        }
        if (pseudo === '' || name === '' || description === '' || photo === '') {
            setError('Pseudo, name, description fields cannot be empty.');
            return true;
        }
        return false;
    }

    const checkForPasswordStrength = () => {
        // check if they match
        console.log(password);
        console.log(validatePassword);
        if (password !== validatePassword) {
            setError('Passwords do not match.');
            return true;
        }
        // at least one number, one lowercase and one uppercase letter and at least six characters
        re = /(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).{6,}/;
        console.log(re.test(password));
        if (!re.test(password)) {
            setError('Password must contain at least one number, one lowercase and one uppercase letter and at least six characters.');
            return true;
        }
        return false;
    }

    const checkForEmailFormat = () => {
        // Email format validation
        var re = /\S+@\S+\.\S+/;
        if (!re.test(email)) {
            setError('Email format is invalid.');
            return true;
        }
        return false;
    }

    const checkForPseudoUniqueness = async () => {
        // check if pseudo is unique from firestore
        const db = getFirestore();
        const querySnapshot = await getDocs(query(collection(db, "users"), where("pseudo", "==", pseudo)));
        if (querySnapshot.size > 0) {
            setError('Pseudo already exists.');
            return true;
        }
        return false;
    }


    const onSignUp = async () => {
        setError('');
        // Validate data integrity
        if (checkForEmptyFields() || checkForEmailFormat() || checkForPasswordStrength()) {
            if (error === '')
                setError('Something went wrong. Please try again.');
            return;
        }

        if (await checkForPseudoUniqueness()) {
            return;
        }

        // Create user with email and password if not exists
        const auth = getAuth();

        try {
            const userCredential = await createUserWithEmailAndPassword(auth, email, password);
            // Signed in
            const userId = userCredential.user.uid;
            // upload image to firebase storage and wait for url to be returned
            await uplodProfilePicture(userId);
            //add to firestore
            await addUser(userId);
            // ...
        } catch (error) {
            const errorCode = error.code;
            const errorMessage = error.message;
            setError(error + " " + errorMessage);
            return;
        }
    }

    const imagePicker = async () => {
        let result = await ImagePicker.launchImageLibraryAsync({
          mediaTypes: ImagePicker.MediaTypeOptions.All,
          allowsEditing: true,
          aspect: [4, 3],
          quality: 1,
          base64: true,
        });
    
        if (!result.canceled) {
          setPhoto(result.uri);
        }
      };

    const uplodProfilePicture = async (userId) => {
        if (photo === defaultPhoto) {
            return;
        }
        const storage = getStorage();
        const storageRef = ref(storage, 'users/' + userId + '/profilePicture');
        const response = await fetch(photo);
        const blob = await response.blob();
        await uploadBytes(storageRef, blob);
        const url = await getDownloadURL(storageRef);
        setPhoto(url);
    }



    const addUser = async (userId) => {
        const db = getFirestore();
        const docRef = await addDoc(collection(db, "users"), {
          _id: userId,
          pseudo: pseudo,
          name : name,
          photo: photo,
          email: email,
          description : description,
        });
        console.log("Document written with ID: ", docRef.id);
      }

      return (
        <View style={styles.container}>
          <View style={styles.form}>
            <View style={styles.imagePicker}>
              <Image source={{ uri: photo }} style={styles.image} />
              <TouchableOpacity style={styles.imagePickerButton} onPress={imagePicker}>
                <Text style={styles.imagePickerButtonText}>Change Profile Picture</Text>
              </TouchableOpacity>
            </View>
      
            <Text style={styles.sectionTitle}>Account Information</Text>
            <TextInput
              style={[styles.textInput, error.includes('Email') && styles.errorInput]}
              placeholder="Email"
              required
              onChangeText={(email) => setEmail(email)}
            />
            <TextInput
              style={[styles.passwordInput, error.includes('assword') && styles.errorInput]}
              placeholder="Password"
              secureTextEntry={true}
              onChangeText={(pass) => setPassword(pass)}
            />
            <TextInput
              style={[styles.passwordInput, password === validatePassword && styles.validatedInput]}
              placeholder="Validate Password"
              secureTextEntry={true}
              onChangeText={(validatePassword) => setValidatePassword(validatePassword)}
            />
      
            <Text style={styles.sectionTitle}>Profile Information</Text>
            <TextInput
              style={[styles.textInput, error.includes('Pseudo') && styles.errorInput]}
              placeholder="Pseudo"
              onChangeText={(pseudo) => setPseudo(pseudo)}
            />
            <TextInput
              style={[styles.textInput, error.includes('ame') && styles.errorInput]}
              placeholder="Name"
              onChangeText={(name) => setName(name)}
            />
            <TextInput
              style={[styles.textInput, error.includes('escription') && styles.errorInput]}
              placeholder="Description"
              onChangeText={(description) => setDescription(description)}
            />
      
            {error ? <Text style={styles.errorText}>{error}</Text> : null}
            <TouchableOpacity style={styles.button} onPress={() => onSignUp()}>
              <Text style={styles.buttonText}>Sign Up</Text>
            </TouchableOpacity>

            <TouchableOpacity style={styles.login} onPress={() => navigation.navigate('Login')}>
                <Text style = {{color: '#3498DB'}}>Already have an account? Login</Text>
            </TouchableOpacity>
          </View>
        </View>
      );
      
      
}

styles = StyleSheet.create({
    container: {
      flex: 1,
      backgroundColor: '#F7F7F7',
      justifyContent: 'center',
      alignItems: 'center',
    },
    logo: {
      width: 100,
      height: 100,
      marginBottom: 20,
    },
    imagePicker: {
      alignItems: 'center',
      marginBottom: 20,
    },
    imagePickerButton: {
      backgroundColor: '#3498DB',
      borderRadius: 5,
      padding: 10,
    },
    imagePickerButtonText: {
      color: '#FFFFFF',
    },
    form: {
      width: '80%',
    },
    textInput: {
      height: 40,
      borderColor: '#E0E0E0',
      borderWidth: 1,
      marginBottom: 10,
      paddingLeft: 10,
      borderRadius: 5,
      color: '#333',
    },
    passwordInput: {
      height: 40,
      borderColor: '#E0E0E0',
      borderWidth: 1,
      marginBottom: 10,
      paddingLeft: 10,
      borderRadius: 5,
      color: '#333',
    },
    errorInput: {
      borderColor: 'red',
    },
    validatedInput: {
      borderColor: 'green',
    },
    button: {
      backgroundColor: '#3498DB',
      padding: 10,
      borderRadius: 5,
      alignItems: 'center',
    },
    buttonText: {
      color: '#FFFFFF',
    },
    errorText: {
      color: 'red',
      marginBottom: 10,
    },
    image: {
      width: 100,
      height: 100,
      borderRadius: 50,
      marginBottom: 10,
    },
    login: {
        marginTop: 20,
        alignItems: 'center',
    }
  });
  