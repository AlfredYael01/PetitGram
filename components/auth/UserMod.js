import React, { useEffect, useState } from "react";
import {
  TouchableOpacity,
  Text,
  TextInput,
  View,
  StyleSheet,
  Image,
} from "react-native";
import { getAuth } from "firebase/auth";
import {
  getFirestore,
  collection,
  getDocs,
  query,
  where
} from "firebase/firestore";
import { LogBox } from "react-native";
LogBox.ignoreLogs(["is deprecated"]);
import { useDispatch, useSelector } from "react-redux";
import * as ImagePicker from "expo-image-picker";
import { updateUser, uploadProfilePicture } from "../helper/user";

export default UserMod = ({ navigation }) => {
  const auth = getAuth();
  const dispatch = useDispatch();
  const userId = auth.currentUser.uid;
  const [profile, setProfile] = useState("");
  const [photo, setPhoto] = useState();
  const [pseudo, setPseudo] = useState();
  const [name, setName] = useState();
  const [description, setDescription] = useState();
  const [error, setError] = useState("");
  const currentUser = useSelector((state) => state.user.currentUser);


  const getUserInfo = async () => {
    setProfile(currentUser)
    setPhoto(currentUser.photo);
    setPseudo(currentUser.pseudo);
    setName(currentUser.name);
    setDescription(currentUser.description);
  };
  useEffect(() => {
    getUserInfo();
  }, []);
  const imagePicker = async () => {
    let result = await ImagePicker.launchImageLibraryAsync({
      mediaTypes: ImagePicker.MediaTypeOptions.All,
      allowsEditing: true,
      aspect: [1, 1],
      quality: 1,
      base64: true,
    });

    if (!result.canceled) {
      try {
        setPhoto(result.uri);
      } catch (e) {
        console.log(e);
      }
    }
  };

  const uploadPhoto = async (userId) => {
    dispatch(uploadProfilePicture({ photo: photo, userId: userId }));
  };

  const checkForPseudoUniqueness = async () => {
    const db = getFirestore();
    const userRef = collection(db, "users");
    const userQuery = query(userRef, where("pseudo", "==", pseudo));
    const querySnapshot = await getDocs(userQuery);
    if (querySnapshot.size > 0) {
      if (querySnapshot.docs[0].data()._id !== userId) {
        setError("Pseudo already used");
        return true;
      }
    }
    if (pseudo === "" || name === "" || description === "") {
      setError("All fields must be filled");
      return true;
    }
    return false;
  };

  const updateUserInfo = async () => {
    if (await checkForPseudoUniqueness()) {
      return;
    }
    dispatch(
      updateUser({
        pseudo: pseudo,
        name: name,
        description: description,
      })
    );
  };

  return (
    <View style={stylesLog.container}>
      <View style={stylesLog.form}>
        <View style={stylesLog.imagePicker}>
          <Image source={{ uri: photo }} style={stylesLog.image} />
          <TouchableOpacity
            style={styles.imagePickerButton}
            onPress={imagePicker}
            testID="imagePickerButton"
          >
            <Text style={styles.imagePickerButtonText}>
              Change Profile Picture
            </Text>
          </TouchableOpacity>
        </View>

        <Text style={stylesLog.sectionTitle}>Account Information</Text>
        <TextInput
          style={[stylesLog.textInput]}
          value={profile.email}
          editable={false}
        />

        <Text style={stylesLog.sectionTitle}>Profile Information</Text>
        <TextInput
          style={[stylesLog.textInput]}
          placeholder={pseudo}
          value={pseudo}
          onChangeText={(text) => setPseudo(text)}
        />
        <TextInput
          style={[stylesLog.textInput]}
          placeholder={name}
          value={name}
          onChangeText={(text) => setName(text)}
        />
        <TextInput
          style={[stylesLog.textInput]}
          placeholder={description}
          value={description}
          onChangeText={(text) => setDescription(text)}
        />

        <TextInput
          style={[stylesLog.errorInput]}
          placeholder=""
          value={error ? error : ""}
          editable={false}
        />

        <TouchableOpacity
          style={stylesLog.button}
          onPress={async () => {
            // if photo did not change set photoUrl to photo else wait for photoUrl to be set
            setError("");
            if (photo !== currentUser.photo) {
              uploadPhoto(userId);
            }
            await updateUserInfo();
            navigation.navigate("UserProfileScreen");
          }}
          testID="ModUser"
        >
          <Text style={stylesLog.buttonText}>Modify profil</Text>
        </TouchableOpacity>
      </View>
    </View>
  );
};

const stylesLog = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#F7F7F7",
    justifyContent: "center",
    alignItems: "center",
  },
  logo: {
    width: 100,
    height: 100,
    marginBottom: 20,
  },
  imagePicker: {
    alignItems: "center",
    marginBottom: 20,
  },
  imagePickerButton: {
    backgroundColor: "#3498DB",
    borderRadius: 5,
    padding: 10,
  },
  imagePickerButtonText: {
    color: "#FFFFFF",
  },
  form: {
    width: "80%",
  },
  textInput: {
    height: 40,
    borderColor: "#E0E0E0",
    borderWidth: 1,
    marginBottom: 10,
    paddingLeft: 10,
    borderRadius: 5,
    color: "#333",
  },
  errorInput: {
    borderColor: "red",
    color: "lightcoral",
    alignSelf: "center",
  },
  validatedInput: {
    borderColor: "green",
  },
  button: {
    backgroundColor: "#3498DB",
    padding: 10,
    borderRadius: 5,
    alignItems: "center",
  },
  buttonText: {
    color: "#FFFFFF",
  },
  image: {
    width: 150,
    height: 150,
    borderRadius: 50,
    marginBottom: 10,
  },
});
