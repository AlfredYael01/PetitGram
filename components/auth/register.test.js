import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import Register from './Register';
import { getAuth } from 'firebase/auth';
import { getFirestore, collection, getDocs, query, where } from 'firebase/firestore';
import { createUserWithEmailAndPassword } from 'firebase/auth';

// Mock the modules that are not available in the test environment
jest.mock('expo-image-picker', () => ({
    launchImageLibraryAsync: jest.fn(),
    MediaTypeOptions: {
        All: 'All', // Add MediaTypeOptions with 'All' to mock the MediaTypeOptions used in your component
    },
}));

jest.mock('firebase/firestore', () => ({
    getFirestore: jest.fn(),
    collection: jest.fn(),
    getDocs: jest.fn( () => ({
        size: 1,
        docs: [
            {
                data: jest.fn( () => ({
                    pseudo: 'test',
                })),
            },
        ],
    })),
    query: jest.fn(),
    where: jest.fn(),
}));

jest.mock('firebase/auth', () => ({
    getAuth: jest.fn(),
    createUserWithEmailAndPassword: jest.fn(),
}));

jest.mock('firebase/storage', () => ({
    getStorage: jest.fn(),
    ref: jest.fn(),
    uploadBytes: jest.fn(),
    getDownloadURL: jest.fn(),
}));

jest.mock('expo-image-picker', () => ({
    launchImageLibraryAsync: jest.fn(),
    MediaTypeOptions: {
        All: 'All', // Add MediaTypeOptions with 'All' to mock the MediaTypeOptions used in your component
    },
}));


describe('Register Component', () => {
  it('renders correctly', () => {
    const { getByText, getByPlaceholderText, getByTestId } = render(<Register navigation={jest.fn()} />);
    
    // Add assertions for rendering elements and placeholders
    expect(getByText('Sign Up')).toBeTruthy();
    expect(getByPlaceholderText('Email')).toBeTruthy();
    expect(getByPlaceholderText('Password')).toBeTruthy();
    expect(getByPlaceholderText('Validate Password')).toBeTruthy();
    expect(getByPlaceholderText('Pseudo')).toBeTruthy();
    expect(getByPlaceholderText('Name')).toBeTruthy();
    expect(getByPlaceholderText('Description')).toBeTruthy();
    expect(getByTestId('imagePickerButton')).toBeTruthy();
  });

  it('displays error messages for empty fields', async () => {
    const { getByTestId, getByText, getByPlaceholderText } = render(<Register navigation={jest.fn()} />);
    
    // Trigger sign up without filling any fields
    await fireEvent.press(getByTestId('signUpButton'));
    // Add assertions for error messages
    expect(getByText('Email and password fields cannot be empty.')).toBeTruthy();

    // Fill in fields
    await fireEvent.changeText(getByPlaceholderText('Email'), 'test@example.com');
    await fireEvent.changeText(getByPlaceholderText('Password'), 'Password1');
    await fireEvent.changeText(getByPlaceholderText('Validate Password'), 'Password1');

    // Trigger sign up
    await fireEvent.press(getByTestId('signUpButton'));

    expect (getByText('Pseudo, name, description fields cannot be empty.')).toBeTruthy();
  });

  it('displays error message for non-matching passwords', async () => {
    const { getByTestId, getByText, getByPlaceholderText } = render(<Register navigation={jest.fn()} />);
    
    // Fill in fields
    await fireEvent.changeText(getByPlaceholderText('Email'), 'test@example.com');
    await fireEvent.changeText(getByPlaceholderText('Password'), 'Password1');
    await fireEvent.changeText(getByPlaceholderText('Validate Password'), 'Password2');
    await fireEvent.changeText(getByPlaceholderText('Pseudo'), 'test');
    await fireEvent.changeText(getByPlaceholderText('Name'), 'Test User');
    await fireEvent.changeText(getByPlaceholderText('Description'), 'Test Description');
    
    // Trigger sign up
    await fireEvent.press(getByTestId('signUpButton'));
    // Add assertions for the "Passwords do not match" error message
    expect(getByText('Passwords do not match.')).toBeTruthy();
  });

  it('displays error message for invalid email format', async () => {
    const { getByText, getByPlaceholderText } = render(<Register navigation={jest.fn()} />);
    
    // Fill in fields
    await fireEvent.changeText(getByPlaceholderText('Email'), 'invalid-email-format');
    await fireEvent.changeText(getByPlaceholderText('Password'), 'Password1');
    await fireEvent.changeText(getByPlaceholderText('Validate Password'), 'Password1');
    await fireEvent.changeText(getByPlaceholderText('Pseudo'), 'test');
    await fireEvent.changeText(getByPlaceholderText('Name'), 'Test User');
    await fireEvent.changeText(getByPlaceholderText('Description'), 'Test Description');
    
    // Trigger sign up
    await fireEvent.press(getByText('Sign Up'));
    
    // Add assertions for the "Email format is invalid" error message
    expect(getByText('Email format is invalid.')).toBeTruthy();
  });

  it('displays error message for non-unique pseudo', async () => {
    const { getByText, getByPlaceholderText } = render(<Register navigation={jest.fn()} />);
    
    // Fill in fields
    await fireEvent.changeText(getByPlaceholderText('Email'), 'test@example.com');
    await fireEvent.changeText(getByPlaceholderText('Password'), 'Password1');
    await fireEvent.changeText(getByPlaceholderText('Validate Password'), 'Password1');
    await fireEvent.changeText(getByPlaceholderText('Pseudo'), 'test');
    await fireEvent.changeText(getByPlaceholderText('Name'), 'John Doe');
    await fireEvent.changeText(getByPlaceholderText('Description'), 'Description');
    
    // Trigger sign up
    await fireEvent.press(getByText('Sign Up'));
    
    // Add an assertion for the "Pseudo already exists" error message
    expect(getByText('Pseudo already exists.')).toBeTruthy();
  });

  it('successfully registers a new user', async () => {
    // You should mock the Firebase functions used in your component, such as createUserWithEmailAndPassword and addDoc
    // You can use a library like `firebase-mock` for mocking Firebase functionality in your tests.

    const { getByText, getByPlaceholderText } = render(<Register navigation={jest.fn()} />);
    
    // Fill in valid fields
    await fireEvent.changeText(getByPlaceholderText('Email'), 'new_user@example.com');
    await fireEvent.changeText(getByPlaceholderText('Password'), 'NewUser1');
    await fireEvent.changeText(getByPlaceholderText('Validate Password'), 'NewUser1');
    await fireEvent.changeText(getByPlaceholderText('Pseudo'), 'unique_pseudo');
    await fireEvent.changeText(getByPlaceholderText('Name'), 'New User');
    await fireEvent.changeText(getByPlaceholderText('Description'), 'New User Description');

    // change the mock implementation of getDocs to return an empty query snapshot
    getDocs.mockImplementationOnce(() => ({
        size: 0,
        docs: [],
    }));
    
    // Trigger sign up
    await fireEvent.press(getByText('Sign Up'));
    
    // Add an assertion for successful registration
    // You can check if the navigation function is called to navigate to the next screen
    expect(createUserWithEmailAndPassword).toHaveBeenCalledWith(getAuth(), 'new_user@example.com', 'NewUser1');
  });
});
