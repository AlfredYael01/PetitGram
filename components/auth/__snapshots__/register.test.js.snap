// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register component renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#F7F7F7",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      {
        "width": "80%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "marginBottom": 20,
        }
      }
    >
      <Image
        source={
          {
            "uri": "https://firebasestorage.googleapis.com/v0/b/petitgram-b48fd.appspot.com/o/Profile%2FuserImage.png?alt=media&token=29660ffe-caba-4fe6-b028-09af3f446b74&_gl=1*i459ow*_ga*NDMzMjcyMjA3LjE2OTU4ODMxMjk.*_ga_CW55HF8NVT*MTY5OTM0NzExOC4xOS4xLjE2OTkzNDcxNzIuNi4wLjA",
          }
        }
        style={
          {
            "borderRadius": 50,
            "height": 100,
            "marginBottom": 10,
            "width": 100,
          }
        }
      />
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#3498DB",
            "borderRadius": 5,
            "opacity": 1,
            "padding": 10,
          }
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
            }
          }
        >
          Change Profile Picture
        </Text>
      </View>
    </View>
    <Text>
      Account Information
    </Text>
    <TextInput
      onChangeText={[Function]}
      placeholder="Email"
      required={true}
      style={
        [
          {
            "borderColor": "#E0E0E0",
            "borderRadius": 5,
            "borderWidth": 1,
            "color": "#333",
            "height": 40,
            "marginBottom": 10,
            "paddingLeft": 10,
          },
          false,
        ]
      }
    />
    <TextInput
      onChangeText={[Function]}
      placeholder="Password"
      secureTextEntry={true}
      style={
        [
          {
            "borderColor": "#E0E0E0",
            "borderRadius": 5,
            "borderWidth": 1,
            "color": "#333",
            "height": 40,
            "marginBottom": 10,
            "paddingLeft": 10,
          },
          false,
        ]
      }
    />
    <TextInput
      onChangeText={[Function]}
      placeholder="Validate Password"
      secureTextEntry={true}
      style={
        [
          {
            "borderColor": "#E0E0E0",
            "borderRadius": 5,
            "borderWidth": 1,
            "color": "#333",
            "height": 40,
            "marginBottom": 10,
            "paddingLeft": 10,
          },
          {
            "borderColor": "green",
          },
        ]
      }
    />
    <Text>
      Profile Information
    </Text>
    <TextInput
      onChangeText={[Function]}
      placeholder="Pseudo"
      style={
        [
          {
            "borderColor": "#E0E0E0",
            "borderRadius": 5,
            "borderWidth": 1,
            "color": "#333",
            "height": 40,
            "marginBottom": 10,
            "paddingLeft": 10,
          },
          false,
        ]
      }
    />
    <TextInput
      onChangeText={[Function]}
      placeholder="Name"
      style={
        [
          {
            "borderColor": "#E0E0E0",
            "borderRadius": 5,
            "borderWidth": 1,
            "color": "#333",
            "height": 40,
            "marginBottom": 10,
            "paddingLeft": 10,
          },
          false,
        ]
      }
    />
    <TextInput
      onChangeText={[Function]}
      placeholder="Description"
      style={
        [
          {
            "borderColor": "#E0E0E0",
            "borderRadius": 5,
            "borderWidth": 1,
            "color": "#333",
            "height": 40,
            "marginBottom": 10,
            "paddingLeft": 10,
          },
          false,
        ]
      }
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#3498DB",
          "borderRadius": 5,
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
          }
        }
      >
        Sign Up
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "marginTop": 20,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "color": "#3498DB",
          }
        }
      >
        Already have an account? Login
      </Text>
    </View>
  </View>
</View>
`;
